code/obj/item/storage/toolbox.dm

```diff
@@ -64,10 +64,11 @@
 			return
 
 		if (istype(W, /obj/item/storage/toolbox) || istype(W, /obj/item/storage/box) || istype(W, /obj/item/storage/belt))
-			var/obj/item/storage/S = W
-			for (var/obj/item/I in S.get_contents())
-				if (..(I, user, S) == 0)
+			for (var/obj/item/I as anything in W.storage.get_contents())
+				if (src.storage.is_full())
 					break
+				if (src.storage.check_can_hold(I))
+					W.storage.transfer_stored_item(I, src, TRUE, user)
```

Comment:
This just looks like a reimplementation of stuff that `/datum/storage/storage_item_attack_hand()` already handles (ie can probably be deleted)

Reply:
Agree for toolboxes and belts, but box check was there to prevent stuffing boxes into toolboxes. Added something new for this in https://github.com/goonstation/goonstation/pull/13023/commits/a8e5b71a07565e1e2ae5cc900285f4c03fd3d151